qiime/differential_abundance.py

```diff
@@ -19,19 +22,43 @@
 __email__ = "sophie.sjw@gmail.com"
 
 
+def check_mapping_file_category(loaded_biom, mapping_fp, mapping_category, subcategory_1, subcategory_2):
+    #remove mapping file samples that are not in the input BIOM table
+    with open(mapping_fp, 'U') as map_f:
+        md_map = MetadataMap.parseMetadataMap(mapping_fp)
+    md_map.filterSamples(loaded_biom._sample_ids, strict=True)
```

Comment:
It isn't safe to use "private" attributes/methods (i.e., things prefixed with an underscore), please update to:

```python
loaded_biom.ids(axis='sample')
```